plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
I want to point out that we are going to have to move away from endpoints as the "thing to lock on" to a different object, or a more restricted endpoint, that doesn't make the cluster unsafe.

There's no reason that you have to give a relatively low privileged things write access to its endpoints in order to do locking, or let it potentially alter other aspects of endpoints.